src/crimson/admin/admin_socket.cc

```diff
@@ -342,18 +328,16 @@ class HelpHook final : public AdminSocketHook {
 				      std::string_view format,
 				      ceph::bufferlist&&) const final
   {
-    return seastar::with_shared(m_as.servers_tbl_rwlock,
-				[format, this] {
-      unique_ptr<Formatter> f{Formatter::create(format, "json-pretty", "json-pretty")};
-      f->open_object_section("help");
-      for (const auto& [prefix, hook] : m_as) {
-        if (!hook->help.empty()) {
-          f->dump_string(prefix.data(), hook->help);
-	}
+    unique_ptr<Formatter> f{Formatter::create(format,
+					      "json-pretty", "json-pretty")};
+    f->open_object_section("help");
+    for (const auto& [prefix, hook] : m_as) {
+      if (!hook->help.empty()) {
+        f->dump_string(prefix.data(), hook->help);
       }
-      f->close_section();
-      return seastar::make_ready_future<tell_result_t>(std::move(f));
-    });
+    }
+    f->close_section();
+    return seastar::make_ready_future<tell_result_t>(std::move(f));
   }
```

Comment:
can we make this not return future?

Reply:
please note, `call()` is a virtual function. we could have an extra layer allowing an async `call()` calls into the synchronous version, and convert the plain return value into a `seastar::future<>`. but i want to make this simple.